kolibri/plugins/document_pdf_render/assets/src/views/pdfPage.vue

```diff
@@ -43,27 +43,41 @@
         type: Number,
         required: true,
       },
-      defaultHeight: {
+      firstPageHeight: {
         type: Number,
         required: true,
       },
-      defaultWidth: {
+      firstPageWidth: {
         type: Number,
         required: true,
       },
     },
     data: () => ({
-      height: null,
-      width: null,
       canvas: null,
       rendered: false,
     }),
     computed: {
-      pageHeight() {
-        return this.height || this.defaultHeight;
+      actualHeight() {
+        if (this.pageReady) {
+          return this.pdfPage.view[3];
+        }
+      },
+      actualWidth() {
+        if (this.pageReady) {
+          return this.pdfPage.view[2];
+        }
```

Comment:
It's good to explicitly return `null` so the behavior looks intentional and well-defined.

In cases like this, I personally like to check for the exceptional case rather than the primary case so that the primary logic isn't indented:

```javascript
actualWidth() {
  if (!this.pageReady) {
    return null;
  }
  return this.pdfPage.view[2];
}
```